core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java

```diff
@@ -1743,8 +1746,17 @@ public String getSubclassForDiscriminatorValue(Object value) {
 		return spaces;
 	}
 
-	public TupleContext getTupleContext() {
-		return tupleContext;
+	/**
+	 * Returns the {@link TupleContext}.
+	 *
+	 * @param session the current session, if null the {@link TupleContext#getTransactionContext()} will be null.
+	 * @return the tupleContext for the session
+	 */
+	public TupleContext getTupleContext( SessionImplementor session ) {
```

Comment:
It happens when massindexing, in that case there is not a session when the foreach method is called.
At the moment, Neo4j does not work with the massindexer because of different Lucene versions. 
It should be possible to implement it in the remote scenario but I haven't done that yet